### PR TITLE
Move current feature to _old package and introduce new Mobius constructor to build logic for edit patient screen

### PR DESCRIPTION
**Pull request by ragunathjawahar**

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientUi.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package org.simple.clinic.editpatient
+
+import org.simple.clinic.editpatient_old.PatientEditValidationError
+import org.simple.clinic.patient.Gender
+import org.simple.clinic.widgets.ageanddateofbirth.DateOfBirthAndAgeVisibility
+import org.threeten.bp.LocalDate
+
+interface EditPatientUi {
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/editpatient/EditPatientUi.kt]: Shouldn't this be in the `_old` package?

Comment (ragunathjawahar) [reply app/src/main/java/org/simple/clinic/editpatient/EditPatientUi.kt]: I'm reusing classes when possible.

File: mobius-migration/src/main/java/org/simple/mobius/migration/MobiusTestFixture.kt [hunk 2]
```diff
@@ -38,7 +39,15 @@ class MobiusTestFixture<M: Any, E, F>(
       effectHandler: EffectHandler<F, E>,
       modelUpdateListener: ModelUpdateListener<M>,
       requiresLogging: Boolean = false
-  ) : this(events, defaultModel, initFunction, Update { model: M, event: E -> updateFunction(model, event) }, effectHandler, modelUpdateListener, requiresLogging)
+  ) : this(
+      events,
+      defaultModel,
+      Init<M, F> { initFunction?.invoke(it) ?: First.first(it) },
```
Comment (vinaysshenoy) [inline mobius-migration/src/main/java/org/simple/mobius/migration/MobiusTestFixture.kt]: I checked just now, and it seems like explicitly defining the type as `<M, F>` is unnecessary and we can just leave it as `Init {}`. Please check?

Comment (ragunathjawahar) [reply mobius-migration/src/main/java/org/simple/mobius/migration/MobiusTestFixture.kt]: Fixed!

**Comment by ragunathjawahar** [comment]
@vinaysshenoy force pushed hence your previous approval became stale after resolving conflicts. Can you please approve again?

**Pull request merged**